bom-decomposer/src/main/java/io/quarkus/bom/decomposer/PomUtils.java

```diff
@@ -86,6 +76,42 @@ public static Model toModel(DecomposedBom decomposed, Model baseModel) {
         return model;
     }
 
+    private static Dependency toModelDep(ProjectDependency dep) {
+        final org.eclipse.aether.graph.Dependency aetherDep = dep.dependency();
+        final Artifact a = dep.artifact();
+
+        final Dependency modelDep = new Dependency();
+        modelDep.setGroupId(a.getGroupId());
+        modelDep.setArtifactId(a.getArtifactId());
+        if (!a.getClassifier().isEmpty()) {
+            modelDep.setClassifier(a.getClassifier());
+        }
+        if (!"jar".equals(a.getExtension())) {
+            modelDep.setType(a.getExtension());
+        }
+        modelDep.setVersion(a.getVersion());
+
+        final String scope = aetherDep.getScope();
+        if (!scope.isBlank() && !"compile".equals(scope)) {
+            modelDep.setScope(scope);
+        }
+
+        final Boolean optional = aetherDep.getOptional();
+        if (optional != null && optional) {
+            modelDep.setOptional(true);
+        }
+
+        if (!aetherDep.getExclusions().isEmpty()) {
+            for (org.eclipse.aether.graph.Exclusion aetherExcl : aetherDep.getExclusions()) {
+                final Exclusion modelExcl = new Exclusion();
+                modelExcl.setGroupId(aetherExcl.getGroupId());
+                modelExcl.setArtifactId(aetherExcl.getArtifactId());
```

Comment:
Don't you need the classifier too if there is one?

Reply:
It does not support classifiers in the pom.